ruby/example_code/iam/scenario_create_user_assume_role.rb

```diff
@@ -0,0 +1,295 @@
+# 5. Delete the policy, role, and user.
+
+# snippet-start:[ruby.example_code.iam.Scenario_CreateUserAssumeRole]
+require 'aws-sdk-iam'
+require 'aws-sdk-s3'
+
+# Wraps the scenario actions.
+class ScenarioCreateUserAssumeRole
+  attr_reader :iam_resource
+
+  # @param iam_resource [Aws::IAM::Resource] An AWS IAM resource.
+  def initialize(iam_resource)
+    @iam_resource = iam_resource
+  end
+
+  # Waits for the specified number of seconds.
+  #
+  # @param duration [Integer] The number of seconds to wait.
+  def wait(duration)
+    puts("Give AWS time to propagate resources...")
+    sleep(duration)
+  end
+
+  # snippet-start:[ruby.example_code.iam.CreateUser]
+  # Creates a user.
+  #
+  # @param user_name [String] The name to give the user.
+  # @return [Aws::IAM::User] The newly created user.
+  def create_user(user_name)
+    user = @iam_resource.create_user(user_name: user_name)
+    puts("Created demo user named #{user.name}.")
+  rescue Aws::Errors::ServiceError => e
+    puts('Tried and failed to create demo user.')
+    puts("\t#{e.code}: #{e.message}")
+    puts("\nCan't continue the demo without a user!")
+    raise
+  else
+    user
+  end
+  # snippet-end:[ruby.example_code.iam.CreateUser]
+
+  # snippet-start:[ruby.example_code.iam.CreateAccessKey]
+  # Creates an access key for a user.
+  #
+  # @param user [Aws::IAM::User] The user that owns the key.
+  # @return [Aws::IAM::AccessKeyPair] The newly created access key.
+  def create_access_key_pair(user)
+    user_key = user.create_access_key_pair
+    puts("Created access key pair for user.")
+  rescue Aws::Errors::ServiceError => e
+    puts("Couldn't create access keys for user #{user.name}.")
+    puts("\t#{e.code}: #{e.message}")
+    raise
+  else
+    user_key
+  end
+  # snippet-end:[ruby.example_code.iam.CreateAccessKey]
+
+  # snippet-start:[ruby.example_code.iam.CreateRole]
+  # Creates a role that can be assumed by a user.
+  #
+  # @param role_name [String] The name to give the role.
+  # @param user [Aws::IAM::User] The user who is granted permission to assume the role.
+  # @return [Aws::IAM::Role] The newly created role.
+  def create_role(role_name, user)
+    role = @iam_resource.create_role(
+      role_name: role_name,
+      assume_role_policy_document: {
+        Version: '2012-10-17',
+        Statement: [{
+          Effect: 'Allow',
+          Principal: {'AWS': user.arn},
+          Action: 'sts:AssumeRole'
+        }]
+      }.to_json)
+    puts("Created role #{role.name}.")
+  rescue Aws::Errors::ServiceError => e
+    puts("Couldn't create a role for the demo. Here's why: ")
+    puts("\t#{e.code}: #{e.message}")
+    raise
+  else
+    role
+  end
+  # snippet-end:[ruby.example_code.iam.CreateRole]
+
+  # snippet-start:[ruby.example_code.iam.CreatePolicy]
+  # snippet-start:[ruby.example_code.iam.AttachRolePolicy]
+  # Creates a policy that grants permission to list buckets in the account and attaches
+  # it to a role.
```

Comment:
Creates a policy that grants permission to list S3 buckets in the account, and then attaches the policy to a role.

Reply:
Fixed